src/code/z_camera.c

```diff
@@ -1,3070 +1,6778 @@
+    cur->z = Camera_LERPCeilF(target->z, cur->z, xzStepScale, minDiff);
 }
 
 void func_80043ABC(Camera* camera) {
-    camera->unk_C0.z = 100.0f;
-    camera->unk_C0.y = OREG(7);
-    camera->unk_C0.x = OREG(6);
-    camera->unk_CC.x = OREG(2) * 0.01f;
-    camera->unk_CC.y = OREG(3) * 0.01f;
-    camera->unk_CC.z = OREG(4) * 0.01f;
+    camera->yawUpdateRateInv = 100.0f;
+    camera->pitchUpdateRateInv = R_CAM_DEFA_PHI_UPDRATE;
+    camera->rUpdateRateInv = OREG(6);
+    camera->xzOffsetUpdateRate = PCT(OREG(2));
+    camera->yOffsetUpdateRate = PCT(OREG(3));
+    camera->fovUpdateRate = PCT(OREG(4));
 }
 
 void func_80043B60(Camera* camera) {
-    camera->unk_C0.x = OREG(27);
-    camera->unk_C0.z = OREG(27);
-    camera->unk_C0.y = OREG(27);
-    camera->unk_CC.x = 0.001f;
-    camera->unk_CC.y = 0.001f;
-    camera->unk_CC.z = 0.001f;
+    camera->rUpdateRateInv = OREG(27);
+    camera->yawUpdateRateInv = OREG(27);
+    camera->pitchUpdateRateInv = OREG(27);
+    camera->xzOffsetUpdateRate = 0.001f;
+    camera->yOffsetUpdateRate = 0.001f;
+    camera->fovUpdateRate = 0.001f;
 }
 
-Vec3f* func_80043BC4(Vec3f* arg0, Vec3s* arg1) {
-    Vec3f sp4;
+Vec3f* Camera_Vec3sToVec3f(Vec3f* dest, Vec3s* src) {
+    Vec3f copy;
 
-    sp4.x = arg1->x;
-    sp4.y = arg1->y;
-    sp4.z = arg1->z;
+    copy.x = src->x;
+    copy.y = src->y;
+    copy.z = src->z;
 
-    *arg0 = sp4;
-    return arg0;
+    *dest = copy;
+    return dest;
 }
 
-Vec3f* func_80043C28(Vec3f* arg0, Vec3f* arg1, VecSph* arg2) {
-    Vec3f sp24, sp18;
-    OLib_VecSphGeoToVec3f(&sp18, arg2);
+Vec3f* Camera_Vec3fVecSphGeoAdd(Vec3f* dest, Vec3f* a, VecSph* b) {
+    Vec3f copy, vecB;
+    OLib_VecSphGeoToVec3f(&vecB, b);
 
-    sp24.x = arg1->x + sp18.x;
-    sp24.y = arg1->y + sp18.y;
-    sp24.z = arg1->z + sp18.z;
+    copy.x = a->x + vecB.x;
+    copy.y = a->y + vecB.y;
+    copy.z = a->z + vecB.z;
 
-    *arg0 = sp24;
-    return arg0;
+    *dest = copy;
+    return dest;
 }
 
-Vec3f* func_80043CAC(Vec3f* arg0, Vec3f* arg1, Vec3f* arg2, f32 arg3) {
-    Vec3f sp4;
+Vec3f* Camera_Vec3fTranslateByUnitVector(Vec3f* dest, Vec3f* src, Vec3f* unitVector, f32 uvScale) {
+    Vec3f copy;
 
-    sp4.x = arg1->x + (arg2->x * arg3);
-    sp4.y = arg1->y + (arg2->y * arg3);
-    sp4.z = arg1->z + (arg2->z * arg3);
+    copy.x = src->x + (unitVector->x * uvScale);
+    copy.y = src->y + (unitVector->y * uvScale);
+    copy.z = src->z + (unitVector->z * uvScale);
 
-    *arg0 = sp4;
-    return arg0;
+    *dest = copy;
+    return dest;
 }
 
-s32 func_80043D18(Camera* camera, Vec3f* arg1, struct_80043D18* arg2) {
+/**
+ * Detects the collision poly between `from` and `to`, places collision info in `to`
+ */
+s32 Camera_BGCheckInfo(Camera* camera, Vec3f* from, CamColChk* to) {
     CollisionContext* colCtx;
-    Vec3f sp68, sp5C, sp50;
-    f32 temp_ret;
-    CollisionPoly* sp48;
-    s32 sp44;
-    VecSph sp3C;
+    Vec3f toNewPos, toPoint, fromToNorm;
+    f32 floorPolyY;
```

Comment:
in general in the codebase, we dont tend to do multiple declarations on one line.
I think later on we can consider it when it makes sense? but for now in the decomp stage while we may need to swap declaration orders around itd be best to list all of them individually.
espeically cause its hard to be consistent with this at this stage

this is relevant in a few places in the file